src/nexus-prisma/NexusPrismaBuilder.ts

```diff
@@ -112,17 +95,80 @@ export class NexusPrismaBuilder {
               mapping,
             }
           })
-
-          return this.buildSchemaForCRUD(t, 'Mutation', mutationFields)
+        } else {
+          throw new Error(
+            `t.crud can only be used on a 'Query' & 'Mutation' objectType. Please use 't.model' instead`,
+          )
         }
+
+        type FieldPublisher = (opts?: FieldPublisherConfig) => CRUDMethods // Fluent API
+        type CRUDMethods = Record<string, FieldPublisher>
+
+        return mappedFields.reduce<CRUDMethods>((crud, mappedField) => {
+          const prismaModelName = mappedField.mapping.model
+
+          mappedField.fields.forEach(field => {
+            const mappedFieldName = getCRUDFieldName(
+              prismaModelName,
+              field.name,
+              mappedField.mapping,
+              this.fieldNamingStrategy,
+            )
+            const fieldPublisher: FieldPublisher = givenConfig => {
+              const resolvedConfig: FieldPublisherConfig = {
+                pagination: true,
+                type: field.outputType.type,
+                ...givenConfig,
+              }
+              const gqlFieldName = resolvedConfig.alias
+                ? resolvedConfig.alias
+                : mappedFieldName
+              const gqlType = resolvedConfig.type!
+              const operationName = Object.keys(mappedField.mapping).find(
+                key => (mappedField.mapping as any)[key] === field.name,
+              ) as keyof DMMF.Mapping | undefined
+
+              if (!operationName) {
+                throw new Error(
+                  `Could not find operation name for field ${field.name}`,
+                )
+              }
+
+              t.field(gqlFieldName, {
```

Comment:
I removed use of the helper func and spread here. Felt it was easier to read direct usage of the nexus api rather than hiding just a few fields behind spread. We can afford to lose some DRY so long as the type system keeps us honest.

Reply:
Good! I used to have a lot more of these nexus configs in nexus-prisma v1, which lead to some mistake from time to time where I'd forget one field (like `isList`). We don't have that much anymore, so it's fine that way!